snapcraft/parts/setup_assets.py

```diff
@@ -0,0 +1,215 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2022 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Copy assets to their final locations."""
+
+import itertools
+import os
+import shutil
+import stat
+import textwrap
+import urllib.parse
+from pathlib import Path
+from typing import List, Optional
+
+import requests
+from craft_cli import emit
+
+from snapcraft import errors
+from snapcraft.projects import Project
+
+from .desktop_file import DesktopFile
+
+
+def setup_assets(project: Project, *, assets_dir: Path, prime_dir: Path) -> None:
```

Comment:
we should copy hooks here, new PR, we are missing it for some reason or the other...